habitat-hitl/habitat_hitl/core/remote_client_state.py

```diff
@@ -212,11 +213,29 @@ def _update_input_state(self, client_states: List[ClientState]) -> None:
                         continue
                     gui_input._mouse_button_up.add(MouseButton(button))
 
-                delta: List[Any] = mouse_json["scrollDelta"]
-                if len(delta) == 2:
-                    gui_input._mouse_scroll_offset += (
-                        delta[0] if abs(delta[0]) > abs(delta[1]) else delta[1]
-                    )
+                if "scrollDelta" in mouse_json:
+                    delta: List[Any] = mouse_json["scrollDelta"]
+                    if len(delta) == 2:
+                        gui_input._mouse_scroll_offset += (
+                            delta[0]
+                            if abs(delta[0]) > abs(delta[1])
+                            else delta[1]
+                        )
+
+                if "rayOrigin" in mouse_json:
+                    ray_origin: List[float] = mouse_json["rayOrigin"]
+                    ray_direction: List[float] = mouse_json["rayDirection"]
+                    if len(ray_origin) == 3 and len(ray_direction) == 3:
```

Comment:
Is there a case where this is not true? Should there be a warning/error or assertion?

Reply:
Not really, unless another client is developed and the developer did a mistake. A warning would make sense.